dockerclient.go

```diff
@@ -139,8 +144,21 @@ func (client *DockerClient) ListContainers(all bool, size bool, filters string)
 	if size == true {
 		showSize = 1
 	}
+
 	uri := fmt.Sprintf("/%s/containers/json?all=%d&size=%d", APIVersion, argAll, showSize)
 
+	if limit != "" {
+		uri += "&limit=" + limit
```

Comment:
logic like this is a lot nicer using a `url.Values` e.g. https://github.com/samalba/dockerclient/blob/master/dockerclient.go#L628

I'd rather include `all` and `size` in the `url.Values` as well, and move the uri creation to after populating the `url.Values`:

```go
url := fmt.Sprintf("/%s/containers/json?%s", v.Encode())
```